test/test-applications/integrations/Samples.Http/Program.cs

```diff
@@ -0,0 +1,44 @@
+// <copyright file="Program.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Net.Http;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+
+namespace Samples.Http
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var port = TcpPortProvider.GetOpenPort();
+            CreateHostBuilder(args, port).Build().RunAsync();
+
+            using var httpClient = new HttpClient();
+            httpClient.GetAsync($"http://localhost:{port}/test").Wait();
+        }
```

Comment:
should it not be something more or less like

```suggestion
        public static async Task Main(string[] args)
        {
            var port = TcpPortProvider.GetOpenPort();
            var serverTask = CreateHostBuilder(args, port).Build().RunAsync();

            using var httpClient = new HttpClient();
            var clientTask = httpClient.GetAsync($"http://localhost:{port}/test");

            await Task.WhenAll(serverTask, clientTask);
        }
```

?

Reply:
serverTask won't complete in this case, do in the integration test it would hang on `RunSampleAndWaitForExit` method.

Reply:
Is it possible that  `httpClient.GetAsync($"http://localhost:{port}/test")` might be called before the server is listening? I do not want to have another flaky test 😄 If not then it is worth adding a comment e.g. that the server is listening before `RunAsync()` returns.

Reply:
Good point @pellared - I think it should use `Start()` in this case, see https://source.dot.net/#Microsoft.Extensions.Hosting.Abstractions/HostingAbstractionsHostBuilderExtensions.cs,16

The IHost should also be disposed - the program is terminating but it seems reasonable.

Reply:
@dszmigielski you can consider adding a `/shutdown` endpoint that triggers application shutdown